src/main/java/app/persistence/conection/BaseDatos.java

```diff
@@ -84,78 +80,68 @@ public BaseDatos() throws Exception {
     }
 
     private void initEntityManagerFactory() throws Exception {
-        decidirBaseDatos();
-        if (emf == null) {
-            switch (selector) {
-                case LOCAljose:
-                    localMySQlJose();
-                    break;
-                case LOCAljorge:
-                    localMySQljorge();
-                    break;
-                case LOCAlpaolo:
-                    localMySQlpaolo();
-                    break;
-                case LOCALGabi:
-                    localMySQlGabi();
-                    break;
-                case OpenShift:
-                    openShift();
-                    break;
-                case LOCAlfede:
-                    localMySQlFede();
-                case HEROKU:
-                    heroku();
-                    break;
-            }
-        }
+        ConectionMySql cm [] = new ConectionMySql[7];
+        cm[0] = new MySqlJose();
+        cm[1] = new MySqlJorge();
+        cm[2] = new MySqlPaolo();
+        cm[3] = new MySqlOpenShift();
+        cm[4] = new MySqlHeroku();
+        cm[5] = new MySqlFede();
+        cm[6] = new MySqlGabi();
+
+        selector = decidirBaseDatos();
+        BaseDatosSql baseBd = new BaseDatosSql();
+        baseBd.setTypeConection(cm[selector-1]);
+
+        emf =  baseBd.request(emf);
+
     }
 
-    protected EntityManager getEntityManager() {
+    public EntityManager getEntityManager() {
         return em;
     }
 
-    public void decidirBaseDatos() throws Exception {
+    public int decidirBaseDatos() throws Exception {
```

Comment:
Un try catch por clase que te devuelva un boolean.
Y con polimorfismo eliminas el ConectionMySql cm [] 
ya que tener hardocdeado el numero genera una depencencia.

https://stackoverflow.com/questions/126409/ways-to-eliminate-switch-in-code